unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/SlitPlate.cs

```diff
@@ -0,0 +1,302 @@
+                        var generatorGroupTransition = initialPositionLeft + (transition * (float)(slitIndex));
+                        for (var count = 0; count < generatorCountPerSlit; count++)
+                        {
+                            waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].transform.localPosition = new Vector3(generatorGroupTransition + (generatorPlacementTransistion * (count + 1)), left.transform.localPosition.y, left.transform.localPosition.z + 0.02f);                       
+                            if(count == 0)
+                            {
+                                waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].GetComponent<WaveGenerator>().SetPropagationMode(WaveGenerator.WavePropagation.Circular);
+                                if(generatorCountPerSlit > 2)
+                                {
+                                    waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].transform.localPosition = new Vector3(generatorGroupTransition + ((generatorPlacementTransistion * 0.5f) * (count + 1)), left.transform.localPosition.y, left.transform.localPosition.z + 0.02f);
+                                }
+                            }   
+                            else if (count == generatorCountPerSlit - 1)
+                            {
+                                waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].GetComponent<WaveGenerator>().SetPropagationMode(WaveGenerator.WavePropagation.Circular);
+                               
+                                if (generatorCountPerSlit > 2)
+                                {
+                                    waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].transform.localPosition = new Vector3(initialPositionRight - (generatorPlacementTransistion * 0.5f), left.transform.localPosition.y, left.transform.localPosition.z + 0.02f);
+                                }
+                            }
+                        }
+                    }                             
+                }
+            }
+            else
+            {
+                DeactivatePlateWaveGenerators();
+            }
+        }
+
+        private int CalculateGeneratorsPerSlit()
+        {        
+            int numberOfGeneratorsPerSlit = 1 + (int)((slitWidth * 10.0f)/1.5f);
+            return numberOfGeneratorsPerSlit; 
+        }
+
+        public void AddWaveGenerator()
+        {
+            var waveGenerator = new GameObject("WaveGenerator");
+            waveGenerator.transform.parent = GameObject.Find("Plate").transform;
+            waveGenerator.transform.rotation = waveGenerator.transform.parent.rotation;
+            //waveGenerator.transform.localPosition = left.transform.localPosition;
+
+            WaveGenerator waveGeneratorScript = waveGenerator.gameObject.AddComponent<WaveGenerator>();
+            waveGeneratorScript.WaveAmplitude = 0.5f;

```

Comment:
At this point, we have to consider the current amplitude, frequency, and wavelength.